struct.go

```diff
@@ -105,6 +106,12 @@ func structNewIndex(L *lua.LState) int {
 			hint := field.Type().Elem()
 			goValue := lValueToReflect(L, value, hint, nil)
 
+			if !goValue.IsValid() {
+				// Occurs if the assigned value does not match the expected
+				// type for the field
+				L.RaiseError(fmt.Sprintf("could not set field %s: expected type %v", key, hint))
```

Comment:
RaiseError already takes in format string and variadic args.  you can remove the fmt.Sprintf